antarest/study/storage/variantstudy/model/command/update_area_properties.py

```diff
@@ -0,0 +1,123 @@
+#
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+#
+# SPDX-License-Identifier: MPL-2.0
+#
+# This file is part of the Antares project.
+from typing import Any, Dict, List, Optional, Tuple
+
+from typing_extensions import override
+
+from antarest.study.business.model.area_properties_model import (
+    ADEQUACY_PATCH_PATH,
+    OPTIMIZATION_PATH,
+    THERMAL_PATH,
+    AreaPropertiesProperties,
+    AreaPropertiesUpdate,
+)
+from antarest.study.model import STUDY_VERSION_8_3
+from antarest.study.storage.rawstudy.model.filesystem.config.area import (
+    AdequacyPathProperties,
+    OptimizationProperties,
+    ThermalAreasProperties,
+)
+from antarest.study.storage.rawstudy.model.filesystem.config.model import FileStudyTreeConfig
+from antarest.study.storage.rawstudy.model.filesystem.factory import FileStudy
+from antarest.study.storage.variantstudy.model.command.common import CommandName, CommandOutput
+from antarest.study.storage.variantstudy.model.command.icommand import ICommand, OutputTuple
+from antarest.study.storage.variantstudy.model.command_listener.command_listener import ICommandListener
+from antarest.study.storage.variantstudy.model.model import CommandDTO
+
+
+class UpdateAreasProperties(ICommand):
+    """
+    Command used to update multiple areas properties
+    """
+
+    # Overloaded metadata
+    # ===================
+
+    command_name: CommandName = CommandName.UPDATE_AREAS_PROPERTIES
+
+    # Command parameters
+    # ==================
+    properties: Dict[str, AreaPropertiesUpdate]
+
+    @override
+    def _apply_config(self, study_data: FileStudyTreeConfig) -> OutputTuple:
+        updated_areas = ", ".join(self.properties.keys())
+        return CommandOutput(status=True, message=f"Areas properties updated: {updated_areas}"), {}
+
+    @override
+    def _apply(self, study_data: FileStudy, listener: Optional[ICommandListener] = None) -> CommandOutput:
+        for area_id, area_properties in self.properties.items():
+            current_thermal_props, current_optim_properties, current_adequacy_patch = self.fetch_area_properties(
+                study_data, area_id
+            )
+
+            current_properties = AreaPropertiesProperties(
+                thermal_properties=ThermalAreasProperties(**current_thermal_props),
+                optimization_properties=OptimizationProperties(**current_optim_properties),
+                adequacy_properties=AdequacyPathProperties(**current_adequacy_patch),
+            )
+
+            new_properties = current_properties.get_area_properties(area_id=area_id).model_copy(
+                update=area_properties.model_dump(exclude_none=True)
+            )
+
+            current_properties.set_area_properties(area_id, new_properties)
+
+            self.save_area_properties(study_data, area_id, current_properties)
```

Comment:
There is some room for optimization here I think:
the thermal properties file could be udpated only once at the end, instead of read/written once for each area.

Reply:
Yes, done